pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/utils/ServerRequestUtils.java

```diff
@@ -182,12 +186,15 @@ private static void walkStageTree(StageNode node, PinotQuery pinotQuery, TableTy
       pinotQuery.setSelectList(CalciteRexExpressionParser.overwriteSelectList(
           ((ProjectNode) node).getProjects(), pinotQuery));
     } else if (node instanceof AggregateNode) {
-      // set agg list
-      pinotQuery.setSelectList(CalciteRexExpressionParser.addSelectList(pinotQuery.getSelectList(),
-          ((AggregateNode) node).getAggCalls(), pinotQuery));
       // set group-by list
       pinotQuery.setGroupByList(CalciteRexExpressionParser.convertGroupByList(
           ((AggregateNode) node).getGroupSet(), pinotQuery));
+      // set agg list - the elements that are selected for an aggregate query
+      // should include all of the group by list followed by the aggregate functions
+      // if other columns are selected upstream they will be projected out in a
+      // dedicated ProjectNode
+      pinotQuery.setSelectList(CalciteRexExpressionParser.addSelectList(pinotQuery.getGroupByList(),
+          ((AggregateNode) node).getAggCalls(), pinotQuery));
```

Comment:
this is the main issue yes?

Reply:
yes, that's why we couldn't use the `NonAggregationGroupByToDistinctQueryRewriter`. But once we use `NonAggregationGroupByToDistinctQueryRewriter` we need to add `LeafDistinctCompatibilityOperator` to interpret the results.